File: day1/goldStar.py
from collections import Counter

left_list = []
right_list = []

with open('day1/input.txt', 'r') as file:
    for line in file:
        if line.strip():
            left_num, right_num = map(int, line.strip().split())
            left_list.append(left_num)
            right_list.append(right_num)

right_counts = Counter(right_list)

similarity_score = sum(num * right_counts.get(num, 0) for num in left_list)

print(similarity_score)

File: R/day1/goldStar.R
data <- read.table("day1/input.txt", header = FALSE)

left_list <- data$V1
right_list <- data$V2

right_counts <- table(right_list)

similarity_score <- 0

# Calculate similarity score
for (number in left_list) {
  if (number %in% names(right_counts)) {
    similarity_score <- similarity_score + number * right_counts[as.character(number)]
  }
}

cat(similarity_score, "\n")